health-services/resource-estimation-service/src/main/java/org/egov/processor/repository/ServiceRequestRepository.java

```diff
@@ -0,0 +1,68 @@
+package org.egov.processor.repository;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import lombok.extern.slf4j.Slf4j;
+import org.egov.tracer.model.ServiceCallException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Map;
+
+import static org.egov.processor.config.ServiceConstants.EXTERNAL_SERVICE_EXCEPTION;
+import static org.egov.processor.config.ServiceConstants.SEARCHER_SERVICE_EXCEPTION;
+
+@Repository
+@Slf4j
+public class ServiceRequestRepository {
+
+    private ObjectMapper mapper;
+
+    private RestTemplate restTemplate;
+
+
+    @Autowired
+    public ServiceRequestRepository(ObjectMapper mapper, RestTemplate restTemplate) {
+        this.mapper = mapper;
+        this.restTemplate = restTemplate;
+    }
+
+
+    public Object fetchResult(StringBuilder uri, Object request) {
+        mapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+        Object response = null;
+        try {
+            response = restTemplate.postForObject(uri.toString(), request, Map.class);
+        }catch(HttpClientErrorException e) {
+            log.error(EXTERNAL_SERVICE_EXCEPTION,e);
+            throw new ServiceCallException(e.getResponseBodyAsString());
+        }catch(Exception e) {
```

Comment:
Why is an exception not being thrown here?

Reply:
Added code for throwing Exception